pkg/controllers/osc/resources/cloud_provider.go

```diff
@@ -41,3 +48,13 @@ func GetCloudProviderFromMachineDeployment(md *clusterv1alpha1.MachineDeployment
 		Spec: *cloudProvider.CloudProviderSpec,
 	}, nil
 }
+
+func GetCloudConfig(secret *corev1.Secret) (string, error) {
+	cloudConfig := secret.Data[CloudConfigConfigName]
```

Comment:
Please check that you this key actually exists otherwise your could might panic during decoding